component/elasticsearch.libsonnet

```diff
@@ -0,0 +1,127 @@
+local elasticsearch = inv.parameters.openshift4_logging.components.elasticsearch;
+
+
+local machineconfig_journald = [
+  kube._Object('machineconfiguration.openshift.io/v1', 'MachineConfig', '41-' + role + '-journald') {
+    metadata+: {
+      labels+: {
+        'machineconfiguration.openshift.io/role': role,
+      },
+    },
+    spec: {
+      config: {
+        ignition: {
+          version: '2.2.0',
+        },
+        storage: {
+          files: [
+            {
+              contents: {
+                // See https://docs.openshift.com/container-platform/latest/logging/config/cluster-logging-systemd.html
+                source: 'data:text/plain;charset=utf-8;base64,' + std.base64(|||
+                  MaxRetentionSec=1month
+                  RateLimitBurst=10000
+                  RateLimitInterval=1s
+                  Storage=persistent
+                  SyncIntervalSec=1s
+                |||),
+              },
+              filesystem: 'root',
+              mode: 420,
+              path: '/etc/systemd/journald.conf',
+            },
+          ],
+        },
+      },
+    },
+  }
+  for role in [ 'master', 'worker' ]
+];
+
+// Allow cluster-scoped ES operator to access ES pods in openshift-logging
+local netpol_operator = kube.NetworkPolicy('allow-from-openshift-operators-redhat') {
+  spec: {
+    ingress: [
+      {
+        from: [
+          {
+            namespaceSelector: {
+              matchLabels: {
+                name: 'openshift-operators-redhat',
+              },
+            },
+          },
+          {
+            podSelector: {
+              matchLabels: {
+                name: 'elasticsearch-operator',
+              },
+            },
+          },
+        ],
+      },
+    ],
+    podSelector: {},
+    policyTypes: [ 'Ingress' ],
+  },
+};
+
+local kibana_routeToPatch = kube._Object('route.openshift.io/v1', 'Route', 'kibana') {
+  metadata+: {
+    namespace: inv.parameters.openshift4_logging.namespace,
+  },
+};
+
+local kibana_patch = resourceLocker.Patch(kibana_routeToPatch, {
+  spec: {
+    host: elasticsearch.kibana_host,
+  },
+});
+
+// OpenShift has custom RBAC permissions on routes if you want to set a host ┻━┻︵ヽ(`Д´)ﾉ︵ ┻━┻
+local kibana_patchWithAdditionalPermissions = std.map(
+  function(obj)
+    if obj.apiVersion == 'rbac.authorization.k8s.io/v1' && obj.kind == 'Role' then
+      obj {
+        rules+: [
+          {
+            apiGroups: [
+              'route.openshift.io',
+            ],
+            resources: [
+              'routes/custom-host',
+            ],
+            verbs: [
+              '*',
+            ],
+          },
+        ],
+      }
+    else
+      obj
+  , kibana_patch
+);
+
+local alertrules = import 'elastic_alertrules.libsonnet';
+
+// Define outputs below
+if elasticsearch.enabled then
+  {
+    '41_es_machineconfig': machineconfig_journald,
+    '42_es_netpol': netpol_operator,
+    '43_es_alertrules': alertrules.prometheus_rules,
```

Comment:
It's not strictly necessary to have a separate number prefix for each file in the output, I'd suggest just using `40_` for everything generated from this file. That would also address the potential issue with the machine configs I mentioned.